### PR TITLE
Dispatcher updates

### PR DESCRIPTION
**Pull request by kekekeks**
- Fixed Dispatcher.Invoke when called from the UI thread with non-Send priority
- dispatcher operation callbacks now have proper synchronization context that awaits continuations with the same priority
- implemented DispatcherFrame and PushFrame
- implemented ExitAllFrames
- implemented DisableProcessing
- implementing BeginInvokeShutdown

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032260-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032280-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032284-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032296-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Base/Threading/AvaloniaSynchronizationContext.cs [hunk 1]
```diff
@@ -9,6 +10,28 @@ namespace Avalonia.Threading
     /// </summary>
     public class AvaloniaSynchronizationContext : SynchronizationContext
     {
+        internal readonly DispatcherPriority Priority;
+        private readonly NonPumpingLockHelper.IHelperImpl? _nonPumpingHelper =
+            AvaloniaLocator.Current.GetService<NonPumpingLockHelper.IHelperImpl>();
+        
+        public AvaloniaSynchronizationContext():  this(Thread.CurrentThread.GetApartmentState() == ApartmentState.STA)
```
Comment (maxkatz6) [inline src/Avalonia.Base/Threading/AvaloniaSynchronizationContext.cs:17]: I wonder. It is possible to set apartment state on the thread on non-windows platforms (in fact, we set it in our templates). Won't it cause any unnecessary operations here on non-windows?

Comment (maxkatz6) [reply src/Avalonia.Base/Threading/AvaloniaSynchronizationContext.cs:17]: Ok, it seems like IHelperImpl is registered on windows only. And tests.

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**